### PR TITLE
[docc] Fail soundness on docc warnings (bad links), fix a few warnings

### PR DESCRIPTION
**Pull request by ktoso**
- make docc validation check for docc warnings, like dead links
- fix some warnings
- make fishy docs use the `@Comment` again to enable it; (fishy docs is what does our "does this code compile" checks)
  - this is a workaround for rdar://79629220 and https://github.com/apple/swift-markdown/issues/64

### CODE REVIEW

**Comment by ktoso** [comment]
Also resolves https://github.com/apple/swift-distributed-actors/issues/1017

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
@swift-server-bot test this please

File: Sources/DistributedActors/LifecycleMonitoring/LifecycleWatch.swift [hunk 1]
```diff
@@ -29,7 +29,7 @@ public protocol LifecycleWatch: DistributedActor where ActorSystem == ClusterSys
     /// located on has been declared ``Cluster/MemberStatus/down``.
     ///
     /// - Parameter id: the ID of the now terminated actor
-    func terminated(actor id: ActorID) async throws // FIXME(distributed): Should not need to be throwing: https://github.com/apple/swift/pull/59397
```
Comment (ktoso) [inline Sources/DistributedActors/LifecycleMonitoring/LifecycleWatch.swift:32]: hooray! compiler no longer crashes and we can remove this workaround everywhere 🎉 🎉 🎉

File: Sources/DistributedActors/DistributedActors.docc/Receptionist.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        listingTask?.cancel()
+    }
+}
+```
+
+## Topics
+
+### <!--@START_MENU_TOKEN@-->Group<!--@END_MENU_TOKEN@-->
+
+- <!--@START_MENU_TOKEN@-->``Symbol``<!--@END_MENU_TOKEN@-->
```
Comment (ktoso) [inline Sources/DistributedActors/DistributedActors.docc/Receptionist.md:108]: TODO: remove these and

File: Sources/DistributedActors/DistributedActors.docc/Receptionist.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# Receptionist
+
+Discovering actors is an important aspect of distributed programming, as it is _the_ primary way we can discover actors on other nodes,
+and communicate with them.
+
+@Comment { 
+    fishy-docs:enable
+}
```
Comment (ktoso) [inline Sources/DistributedActors/DistributedActors.docc/Receptionist.md:8]: this enables compile testing all code snippets

File: Sources/DistributedActors/WeakActorDictionary.swift [hunk 4]
```diff
@@ -14,8 +14,70 @@
 
 import Distributed
 
-struct WeakActorDictionary {
-    var underlying: [ClusterSystem.ActorID: WeakDistributedActorContainer] = [:]
+/// A dictionary which only weakly retains the
+public struct WeakActorDictionary<Act: DistributedActor>: ExpressibleByDictionaryLiteral
```
Comment (ktoso) [inline Sources/DistributedActors/WeakActorDictionary.swift:18]: TODO: incomplete; need to make them a bit nicer API so we can use them in examples ourselfes as well

**Comment by ktoso** [comment]
Meh sorry this is incomplete I just realized, I'll get through it tomorrow.

**Comment by yim-lee** [comment]
huh, https://ci.swiftserver.group/job/swift-distributed-actors-nightly-prb/564/ was stuck

File: Sources/DistributedActors/DistributedActors.docc/Receptionist.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+The receptionist automatically watches checked-in actors, and removes them from the listing once they have been terminated.
+Other actors which discover the actor, and want to be informed once the actor has terminated, should use the <doc:Lifecycle> APIs.
+
+> Warning: `DistributedReception.Key`s are likely to be collapsed with ``ClusterSystem/ActorID/Metadata-swift.struct`` during the beta releases.
+> See [Make use of ActorTag rather than separate keys infra for reception #950](https://github.com/apple/swift-distributed-actors/issues/950)
+
+### Receptionist: Listings
+
+The opposite side of using a receptionist, is actually obtaining a ``DistributedReceptionist/listing(of:file:line:)`` of actors registered with a specific key.
```
Comment (yim-lee) [inline Sources/DistributedActors/DistributedActors.docc/Receptionist.md:60]:
```suggestion
The opposite of using a receptionist is obtaining a ``DistributedReceptionist/listing(of:file:line:)`` of actors registered with a specific key.
```

File: Sources/DistributedActors/DistributedActors.docc/Receptionist.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+}
+```
+
+A typical pattern to use with listings is to create an unstructured task (using `Task { ... }`),
+and store it inside an actor that will be responsible for interacting with the discovered actors.
+
+Once that actor is deinitialized, that task should be cancelled as well, which we can do in its `deinit`, like this:
+
+```swift
+distributed actor Boss: LifecycleWatch {
```
Comment (yim-lee) [inline Sources/DistributedActors/DistributedActors.docc/Receptionist.md:76]: <3

File: Sources/DistributedActors/Cluster/ClusterEventStream.swift [hunk 7]
```diff
@@ -123,7 +123,6 @@ public struct ClusterEventStream: AsyncSequence {
     }
 }
 
-// FIXME(distributed): the only reason this actor is distributed is because of LifecycleWatch
```
Comment (yim-lee) [inline Sources/DistributedActors/Cluster/ClusterEventStream.swift:126]: This specific actor doesn't need to be DA either and with that we can probably remove `whenLocal` usages too...

Comment (ktoso) [reply Sources/DistributedActors/Cluster/ClusterEventStream.swift:126]: That'd need a way to implement lifecycle watch without DA... we don't have that.

Comment (ktoso) [reply Sources/DistributedActors/Cluster/ClusterEventStream.swift:126]: I'll make a ticket for non-DA lifecycle watching though hm hm

Comment (ktoso) [reply Sources/DistributedActors/Cluster/ClusterEventStream.swift:126]: https://github.com/apple/swift-distributed-actors/issues/1036

**Comment by ktoso** [comment]
Meh, needed a rebase now.

**Pull request merged**